test/internal/laws/Setoid.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var forall = require('./forall');
+
+
+module.exports = forall({
+
+  //  a `equals` a = true
+  reflexivity: function(a) {
+    return a.equals(a);
+  },
+
+  //  a `equals` b = b `equals` a
+  symmetry: function(a, b) {
+    return a.equals(b) === b.equals(a);
+  },
+
+  //  a `equals` b & b `equals` c => a `equals` c
+  transitivity: function(a, b, c) {
+    return !a.equals(b) || !b.equals(c) || a.equals(c);
```

Comment:
`a` and `c` are allowed to be equal even if `b` is different:

```javascript
> const a = 1
> const b = 2
> const c = 1
> a === b && b === c
false
> a === c
true
```

I agree that the current expression takes a few moments to understand, but I like the alternative even less:

```javascript
if (a.equals(b) && b.equals(c)) {
  return a.equals(c);
} else {
  return true;
}
```